### PR TITLE
Mod note improvements

### PR DESCRIPTION
**Pull request by RubyMineshaft**
## Proposed Changes

  - Removes 1024 character limit for displayed messages which was causing some previously readable mod notes to show as `<Message too long>`
  - When annulling games, words "white" and "black" will now be replaced with player links.

### CODE REVIEW

**Comment by BHydden** [comment]
I would be very cautious to blanket replace all usage of the words "black" and "white" with player links...

[further 'reading' :)](https://www.youtube.com/watch?v=CcZdwX4noCE)

**Comment by RubyMineshaft** [comment]
I did think about whether or not it would cause problems but couldn't come up with another use of "white" or "black" in the context of an annulment note.

I suppose we could use something like `#black` and `#white` if there's real concern that we'd run into issues.

**Comment by BHydden** [comment]
I missed that it was isolated to the game file and not all mod notes... you may be right, at this time I can't think of other reasons to use "black" or "white" specifically in an annulment message without intending to refer to the player...

File: src/components/Chat/chat_markup.tsx [hunk 1]
```diff
@@ -187,10 +187,6 @@ const global_replacements: TextReplacement[] = [
 export function chat_markup(body: string, extra_pattern_replacements?: Array<TextReplacement>): Array<JSX.Element> {
     let replacements = global_replacements;
 
-    if (body.length > 1024) {
```
Comment (anoek) [inline src/components/Chat/chat_markup.tsx:190]: I believe this had a purpose, if I recall correctly some game's had exceedingly long messages (put in by bad people), basically they were so long it'd crash some browsers. I'm cool with adding another parameter to this function, but we should have the default check still here as a safe guard.

Comment (RubyMineshaft) [reply src/components/Chat/chat_markup.tsx:190]: What if we add something like a `lengthLimit` parameter which could just be set arbitrarily high for mod notes and kept at 1024 for chat?

Comment (anoek) [reply src/components/Chat/chat_markup.tsx:190]: That sounds great :)

**Comment by RubyMineshaft** [comment]
Should look a bit better now. I added an optional lengthLimit parameter to the chat_markup function, and set the limits for chat back to 1024. 😃

**Comment by anoek** [comment]
Cool, thanks!

**Pull request merged**